libs/langgraph/tests/test_messages_state.py

```diff
@@ -0,0 +1,126 @@
+from uuid import UUID
+
+import pytest
+from langchain_core.messages import (
+    AIMessage,
+    HumanMessage,
+    RemoveMessage,
+    SystemMessage,
+)
+
+from langgraph.graph import add_messages
+
+
+def test_add_single_message():
+    left = [HumanMessage(content="Hello", id="1")]
+    right = AIMessage(content="Hi there!", id="2")
+    result = add_messages(left, right)
+    expected_result = [
+        HumanMessage(content="Hello", id="1"),
+        AIMessage(content="Hi there!", id="2"),
+    ]
+    assert result == expected_result
+
+
+def test_add_multiple_messages():
+    left = [HumanMessage(content="Hello", id="1")]
+    right = [
+        AIMessage(content="Hi there!", id="2"),
+        SystemMessage(content="System message", id="3"),
+    ]
+    result = add_messages(left, right)
+    expected_result = [
+        HumanMessage(content="Hello", id="1"),
+        AIMessage(content="Hi there!", id="2"),
+        SystemMessage(content="System message", id="3"),
+    ]
+    assert result == expected_result
+
+
+def test_update_existing_message():
+    left = [HumanMessage(content="Hello", id="1")]
+    right = HumanMessage(content="Hello again", id="1")
+    result = add_messages(left, right)
+    expected_result = [HumanMessage(content="Hello again", id="1")]
+    assert result == expected_result
+
+
+def test_missing_ids():
+    left = [HumanMessage(content="Hello")]
+    right = [AIMessage(content="Hi there!")]
+    result = add_messages(left, right)
+    assert len(result) == 2
+    assert all(isinstance(m.id, str) and UUID(m.id, version=4) for m in result)
+
+
+def test_remove_message():
```

Comment:
should we also add a test where we're sending multiple `RemoveMessage` w/ the same ID (assuming the issue you ran into)